### DISCUSSION TITLE
MQTT failed to connect: connect ECONNREFUSED

### DISCUSSION THREAD

**Discussion by ripeart** [Koenkk/zigbee2mqtt#11590] [Devices]
_edit: I'm a jackass - I didn't realize I had to have a MQTT broker installed._

Hi, I'm using the Unraid Docker container and am having some authentication issues it looks like. I'm able to set the path to the Zigbee stick without any issues. However when starting the container I am faced with: 

> Zigbee2MQTT:info 2022-02-23 15:22:25: Currently 0 devices are joined:
> Zigbee2MQTT:warn 2022-02-23 15:22:25: `permit_join` set to `true` in configuration.yaml.
> Zigbee2MQTT:warn 2022-02-23 15:22:25: Allowing new devices to join.
> Zigbee2MQTT:warn 2022-02-23 15:22:25: Set `permit_join` to `false` once you joined all devices.
> Zigbee2MQTT:info 2022-02-23 15:22:25: Zigbee: allowing new devices to join.
> Zigbee2MQTT:info 2022-02-23 15:22:25: Connecting to MQTT server at mqtt://192.168.168.100
> Zigbee2MQTT:error 2022-02-23 15:22:25: MQTT failed to connect: connect ECONNREFUSED 192.168.168.100:1883
> 
> Zigbee2MQTT:error 2022-02-23 15:22:25: Exiting...

I have set my Unraid user and password details in the configuration.yaml and I am sure they are correct. Is this the right username and password I should be using or is something else going on which would case the ECONNREFUSED error?

Thanks!

**Comment by lumpynose** [comment]
What mqtt server are you using?  Try configuring it to allow anonymous connections.  I'm using mosquitto and in its config file I have

listener 1883 localhost
listener 1883 192.168.50.3

allow_anonymous true


Then in my zigbee2mqtt configuration.yaml in the mqtt section I have

mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://localhost

**Comment by topperwh** [comment]
I use homebridge, but I found the same problem. 

On the pi MQTT broker, set "allow_anonymous true" in your mosquitto.conf file, it allows clients to connect to the MQTT broker without any authentication. 

use the /var/log/mosquitto/mosquitto.log  to check if they are connect.

Just keep in mind that allowing anonymous access to your broker is not recommended in production environments. It may be useful for testing and development purposes though.
